spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/command/ConfigCommands.java

```diff
@@ -121,115 +90,24 @@ public String target(
 			password = userInput.prompt("Password", "", false);
 		}
 
-		try {
-			this.targetHolder.setTarget(new Target(uri, username, password, skipSslValidation));
-
-			final HttpClientConfigurer httpClientConfigurer = HttpClientConfigurer.create()
-					.targetHost(this.targetHolder.getTarget().getTargetUri())
-					.skipTlsCertificateVerification(skipSslValidation);
-			if (StringUtils.hasText(username) && StringUtils.hasText(password)) {
-				httpClientConfigurer.basicAuthCredentials(username, password);
-			}
-			if (StringUtils.hasText(credentialsProviderCommand)) {
-				this.targetHolder.getTarget().setTargetCredentials(new TargetCredentials(true));
-				final Resource credentialsResource = new ProcessOutputResource(
-						credentialsProviderCommand.split("\\s+"));
-				httpClientConfigurer.addInterceptor(new ResourceBasedAuthorizationInterceptor(credentialsResource));
-			}
-			this.restTemplate.setRequestFactory(httpClientConfigurer.buildClientHttpRequestFactory());
-
-			this.skipperClient = new DefaultSkipperClient(targetHolder.getTarget().getTargetUri().toURL().toString(),
-					this.restTemplate);
-
-			// Other commands will be notified of the new SkipperClient object
-			applicationEventPublisher.publishEvent(new SkipperClientUpdatedEvent(skipperClient));
-
-			// TODO - note, we don't yet know if we can access the specified URI
-			this.targetHolder.getTarget()
-					.setTargetResultMessage(String.format("Successfully targeted %s", uri));
+		this.targetHolder.changeTarget(new Target(uri, username, password, skipSslValidation),
+				credentialsProviderCommand);
 
-		}
-		catch (Exception e) {
-			this.targetHolder.getTarget().setTargetException(e);
-			// TODO do we really want to pass around null?
-			applicationEventPublisher.publishEvent(new SkipperClientUpdatedEvent(null));
-			handleTargetException(this.targetHolder.getTarget());
-		}
 		return (this.targetHolder.getTarget().getTargetResultMessage());
 	}
 
 	@ShellMethod(key = "skipper config info", value = "Show the Skipper server being used.")
 	public AboutInfo info() {
 		Target target = targetHolder.getTarget();
 		if (target.getTargetException() != null) {
-			handleTargetException(target);
 			throw new SkipperServerException(this.targetHolder.getTarget().getTargetResultMessage());
 		}
 		return this.skipperClient.getAboutInfo();
 	}
 
-	private void handleTargetException(Target target) {
```

Comment:
This got removed entirely and can be re-instanciated in TargetHolder. I believe it is not necessary though, thanks to the `stacktrace` command.

Reply:
There prob isn't a need.  This is what happens after shutting down the server.  Seems clear enough.

```skipper:>skipper config info
org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:7577/about": Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused); nested exception is org.apache.http.conn.HttpHostConnectException: Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
Details of the error have been omitted. You can use the stacktrace command to print the full stacktrace.
skipper:>stacktrace 
org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:7577/about": Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused); nested exception is org.apache.http.conn.HttpHostConnectException: Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:666)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:613)
	at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:287)
	at org.springframework.cloud.skipper.client.DefaultSkipperClient.getAboutInfo(DefaultSkipperClient.java:82)
	at org.springframework.cloud.skipper.shell.command.ConfigCommands.info(ConfigCommands.java:105)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:216)
	at org.springframework.shell.Shell.evaluate(Shell.java:169)
	at org.springframework.shell.Shell.run(Shell.java:133)
	at org.springframework.shell.jline.DefaultShellApplicationRunner.run(DefaultShellApplicationRunner.java:78)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:767)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:757)
	at org.springframework.boot.SpringApplication.afterRefresh(SpringApplication.java:747)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1162)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1151)
	at org.springframework.cloud.skipper.shell.ShellApplication.main(ShellApplication.java:32)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51)
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:159)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:359)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:89)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:53)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:652)
	... 27 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:75)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 40 more
```

Reply:
However!  If you start the server on say port 9090, you get that long stack trace on startup which isn't nice.  Give a try.

Reply:
```java -jar spring-cloud-skipper-shell/target/spring-cloud-skipper-shell-1.0.0.BUILD-SNAPSHOT.jar
  ____             _                ____ _                 _
 / ___| _ __  _ __(_)_ __   __ _   / ___| | ___  _   _  __| |
 \___ \| '_ \| '__| | '_ \ / _` | | |   | |/ _ \| | | |/ _` |
  ___) | |_) | |  | | | | | (_| | | |___| | (_) | |_| | (_| |
 |____/| .__/|_|  |_|_| |_|\__, |  \____|_|\___/ \__,_|\__,_|
  ____ |_|   _             |___/
 / ___|| | _(_)_ __  _ __   ___ _ __
 \___ \| |/ / | '_ \| '_ \ / _ \ '__|
  ___) |   <| | |_) | |_) |  __/ |
 |____/|_|\_\_| .__/| .__/ \___|_|
              |_|   |_|
 (v1.0.0.BUILD-SNAPSHOT)


WARN : org.springframework.boot.autoconfigure.mustache.MustacheAutoConfiguration : Cannot find template location: classpath:/templates/ (please add some templates, check your Mustache configuration, or set spring.mustache.check-template-location=false)
org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:7577/about": Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused); nested exception is org.apache.http.conn.HttpHostConnectException: Connect to localhost:7577 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:666)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:613)
	at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:287)
	at org.springframework.cloud.skipper.client.DefaultSkipperClient.getAboutInfo(DefaultSkipperClient.java:82)
	at org.springframework.cloud.skipper.shell.command.support.TargetHolder.attemptConnection(TargetHolder.java:92)
	at org.springframework.cloud.skipper.shell.command.support.TargetHolder.changeTarget(TargetHolder.java:65)
	at org.springframework.cloud.skipper.shell.config.ShellConfiguration$InitialConnectionApplicationRunner.run(ShellConfiguration.java:89)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:767)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:757)
	at org.springframework.boot.SpringApplication.afterRefresh(SpringApplication.java:747)
```